Jasmine/Common/ViewModels/CountDownTimer.swift

```diff
@@ -2,17 +2,21 @@ import Foundation
 
 /// A timer to handle game time
 class CountDownTimer {
-
-    /// Specifies the time remaining in the game.
-    private(set) var timeRemaining: TimeInterval
-
     /// Specifies the total time allowed in the game.
     let totalTimeAllowed: TimeInterval
-
-    /// Retrieves time elapsed since game started
-    var timeElapsed: TimeInterval {
-        return totalTimeAllowed - timeRemaining
-    }
+    /// Specifies the time remaining in the game.
+    private(set) var timeRemaining: TimeInterval
+    /// The timer of this countdown timer.
+    private var timer: Timer?
+
+    /// Function that is fired when timer is started.
+    var timerDidStart: (() -> Void)?
```

Comment:
Can you make one function that accepts an enum of the four conditions? Name the enum `TimerStatus`